test/blackbox/common/DDSBlackboxTestsFindTopic.cpp

```diff
@@ -0,0 +1,386 @@
+{
+    /**
+     * A mock type support class.
+     */
+    struct TestType : public TopicDataType
+    {
+        bool serialize(
+                void*,
+                fastrtps::rtps::SerializedPayload_t*) override
+        {
+            return true;
+        }
+
+        bool deserialize(
+                fastrtps::rtps::SerializedPayload_t*,
+                void*) override
+        {
+            return true;
+        }
+
+        std::function<uint32_t()> getSerializedSizeProvider(
+                void*) override
+        {
+            return {};
+        }
+
+        void* createData() override
+        {
+            return nullptr;
+        }
+
+        void deleteData(
+                void*) override
+        {
+        }
+
+        bool getKey(
+                void*,
+                fastrtps::rtps::InstanceHandle_t*,
+                bool) override
+        {
+            return false;
+        }
+
+    };
+
+public:
+
+    void SetUp() override
+    {
+        // No need to enable the participant, as it does not need to transmit information.
+        DomainParticipantFactoryQos factory_qos;
+        factory_qos.entity_factory().autoenable_created_entities = false;
+        participant_ = DomainParticipantFactory::get_instance()->create_participant(0, PARTICIPANT_QOS_DEFAULT);
+        ASSERT_NE(nullptr, participant_);
+
+        type_.reset(new TestType);
+        type_->setName(c_type_name);
+        type_.register_type(participant_);
+    }
+
+    void TearDown() override
+    {
+        if (nullptr != participant_)
+        {
+            participant_->delete_contained_entities();
+            DomainParticipantFactory::get_instance()->delete_participant(participant_);
+        }
+    }
+
+protected:
+
+    static constexpr const char* const c_type_name = "testing::find_topic::test_type";
+
+    DomainParticipant* participant_ = nullptr;
+    TypeSupport type_;
+    TopicListener listener_;
+
+    /**
+     * Create a topic with parameters different to the default ones, in order to check that the topics returned by
+     * find_topic are not default created.
+     */
+    Topic* create_test_topic()
+    {
+        TopicQos qos;
+        qos.transport_priority().value = 10;
+        return participant_->create_topic(TEST_TOPIC_NAME, c_type_name, qos, &listener_, StatusMask::none());
+    }
+
+    void check_topics(
+            const Topic* lhs,
+            const Topic* rhs)
+    {
+        EXPECT_NE(lhs, rhs);
+        EXPECT_EQ(lhs->get_name(), rhs->get_name());
+        EXPECT_EQ(lhs->get_type_name(), rhs->get_type_name());
+        EXPECT_EQ(lhs->get_qos(), rhs->get_qos());
+        EXPECT_EQ(lhs->get_listener(), rhs->get_listener());
+        EXPECT_EQ(lhs->get_status_mask(), rhs->get_status_mask());
+    }
+
+};
+
+/**
+ * Implements test DDS-DP-FT-01 from the test plan.
+ *
+ * Check timeout behavior of DomainParticipant::find_topic.
+ */
+TEST_F(DDSFindTopicTest, find_topic_timeout)
```

Comment:
I think we can make this test way faster

Reply:
I think it should be at least one scheduler quantum. How about 50 ms?